services/galley/src/Galley/API/MLS/Welcome.hs

```diff
@@ -78,33 +87,34 @@ welcomeRecipients ::
      ]
     r =>
   Welcome ->
-  Sem r [Qualified (UserId, ClientId)]
+  Sem r [(Qualified (UserId, ClientId), Qualified ConvId)]
 welcomeRecipients =
-  traverse
-    ( fmap cidQualifiedClient
-        . derefKeyPackage
-        . gsNewMember
-    )
-    . welSecrets
+  traverse (either throw pure <=< resolveMember . gsNewMember) . welSecrets
+
+resolveMember :: Member BrigAccess r => KeyPackageRef -> Sem r (Either (Tagged 'MLSKeyPackageRefNotFound ()) (Qualified (UserId, ClientId), Qualified ConvId))
```

Comment:
Can you please change the type of the function such that the value on the left of the return type is instead an error effect?